### PR TITLE
feat: refactor system token manager

### PR DESCRIPTION
**Pull request by energyGiver**
### Summary
- Refactor SystemTokenManager pallet .
- AssetLinkInterface is added to AssetLink pallet.

### Describe your changes
- Some AssetLink calls are replaced with methods of AssetLinkInterface. It is included in calls of Assets pallet.
    - Below call includes link_system_token()
        ```rust
        pallet_assets::Call::<T>::force_create_with_metadata() // 
        ```
    - Below call includes unlink_system_token()
        ```rust
        pallet_assets::Call::<T>::set_sufficient_with_unlink_system_token(, false)
        ```
- The multilocation is changed to the SystemTokenId type in the link_system_token() in order to resolve dependency problems at infra-substrate.
    ```rust
    pub fn link_system_token(
    ...
    system_token_id: SystemTokenId,  // asset_multi_location: MultiLocation
    ...
    ```

### Related Issue
* #49 

### Checklist before requesting a review
- [x] I have performed a self-review of my code
- [ ] If it is a core feature, I have added thorough tests.
- [ ] Well documented code
- [x] `cargo clippy`
- [x] `cargo-nightly fmt`

### CODE REVIEW

**Review by cocoyoon — APPROVED** [review]

File: runtime/parachains/src/system_token_manager.rs [hunk 1]
```diff
@@ -718,15 +716,7 @@ pub mod pallet {
...
-			Self::call_dmp_asset_link(
-				&asset_link_call,
-				wrapped_asset_link_pallet_id,
-				wrapped_system_token_id,
-				system_token_id,
-			);
-
-			// para which issued original STI -> para which wanna use wrapped STI
+			// para which issued original STI
 			Self::call_dmp_asset_link(
```
Comment (sweatpotato13) [inline runtime/parachains/src/system_token_manager.rs:720]:
```rust
			Self::call_dmp_asset_link(
				&asset_link_call,
				original_asset_link_pallet_id,
				system_token_id,
				wrapped_system_token_id,
			);
```
`wrapped_system_token` 이 2개 이상일 경우에 `link_system_token` call 에서 아래 `ensure`에 걸리진 않나요?

```rust
			// verify asset is not yet registered
			ensure!(
				!AssetIdMultiLocation::<T>::contains_key(asset_id),
				Error::<T>::AssetAlreadyLinked
			);
```

**Review by sweatpotato13 — COMMENTED** [review]
🐧

File: runtime/parachains/src/system_token_manager.rs (hunk 1 above)
Comment (energyGiver) [reply runtime/parachains/src/system_token_manager.rs:720]: Right, good point! Refactoring should be needed to store all wrapped system tokens linked the local asset. Given the complexity of the feature you mentioned, I will incorporate it into the next PR.

**Review by sweatpotato13 — APPROVED** [review]
👍 
Comment한 부분은 이후 작업에서 반영 해주세요

**Pull request merged**